src/ui/public/directives/saved_object_finder.js

```diff
@@ -49,6 +49,38 @@ module.directive('savedObjectFinder', function ($location, $injector, kbnUrl, Pr
       filterResults();
 
       /**
+       * Boolean that keeps track of whether hits are sorted ascending (true)
+       * or descending (false) by title
+       * @type {Boolean}
+       */
+      self.isAscending = true;
+
+      /**
+       * Sorts saved object finder hits either ascending or descending
+       * @param  {Array} hits Array of saved finder object hits
+       * @return {Array} Array sorted either ascending or descending
+       */
+      self.sortHits = function (hits) {
+        self.isAscending = !self.isAscending;
+
+        if (hits.length) {
+          if (self.isAscending) {
```

Comment:
replace all of this with `return self.isAscending ? _.sortBy(hits, 'title') : _.sortBy(hits, 'title').reverse();`